### PR TITLE
Override IIS webapp name only if package contains param file.

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/Utility.ps1 [hunk 1]
```diff
@@ -170,6 +195,34 @@ function Run-Command
 
 }
 
+function Get-CommandOutput
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/Utility.ps1]: Isn't there any tools in task lib that we could make use of?

File: Tasks/AzureRmWebAppDeployment/Utility.ps1 [hunk 2]
```diff
@@ -145,6 +149,27 @@ function Get-MsDeployCmdArgs
     return $msDeployCmdArgs
 }
 
+function Contains-ParamFile( [String][Parameter(Mandatory=$true)] $packageFile )
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/Utility.ps1]: Can you add trace log, to help us if we have to debug on this.

File: Tasks/AzureRmWebAppDeployment/Utility.ps1 (hunk 1 above)
Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/Utility.ps1]: Task lib has this method -> [invoke-vststool](https://github.com/Microsoft/vsts-task-lib/blob/master/powershell/Docs/Commands.md#invoke-vststool) which sets one environment variable with status code but doesn't gives back result of execution.

File: Tasks/AzureRmWebAppDeployment/Utility.ps1 [hunk 3]
```diff
@@ -145,6 +149,33 @@ function Get-MsDeployCmdArgs
...
+{
+    $msDeployExePath = Get-MsDeployExePath
+
+    $msDeployCheckParamFileCmdArgs = " -verb:getParameters -source:package='" + $packageFile + "'";
+
+    $msDeployCheckParamFileCmd = "`"$msDeployExePath`" $msDeployCheckParamFileCmdArgs"
+
+    Write-Host (Get-VstsLocString -Key "Runningmsdeploycommandtocheckifpackagecontainsparamfile0" -ArgumentList $msDeployCheckParamFileCmd)
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/Utility.ps1]: Make it debug log

File: Tasks/AzureRmWebAppDeployment/Utility.ps1 [hunk 4]
```diff
@@ -100,13 +101,16 @@ function Get-MsDeployCmdArgs
...
-    {
-        $msDeployCmdArgs += [String]::Format(' -setParam:name="IIS Web Application Name",value="{0}"', $webAppNameForMSDeployCmd)
+        if($virtualApplication)
+        {
+            $msDeployCmdArgs += [String]::Format(' -setParam:name="IIS Web Application Name",value="{0}/{1}"', $webAppNameForMSDeployCmd, $virtualApplication)
+        }
+        else
+        {
+            $msDeployCmdArgs += [String]::Format(' -setParam:name="IIS Web Application Name",value="{0}"', $webAppNameForMSDeployCmd)
+        }
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/Utility.ps1]: If no parameters.xml file we are not deploying to virtual application

**Comment by mvvsubbu** [comment]
Looks fine

**Pull request merged**